charts/datadog/templates/otel-configmap.yaml

```
@@ -0,0 +1,65 @@
+{{- if eq (include "should-enable-otel-agent" .Values) "true" }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "agents-install-otel-configmap-name" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{ include "datadog.labels" . | nindent 4 }}
+  annotations:
+    checksum/confd-config: {{ tpl (toYaml .Values.datadog.confd) . | sha256sum }}
+    checksum/autoconf-config: {{ tpl (toYaml .Values.datadog.autoconf) . | sha256sum }}
+data:
+  otel-config.yaml: {{- if .Values.datadog.otelCollector.config }} {{ toYaml .Values.datadog.otelCollector.config | indent 4 }}
+  {{- else }} |
+    receivers:
```

imo to be able to create the checksum easily for the content of the `otel-config.yaml`. you should create a fonction help that will generate the content. So you can sue the same helper fonction as source for the hash function

```yaml
  annotations:
    checksum-otel-config:  {{(include "otel-config-content" .) | sha256sum  }}
```

and in this file 

```suggestion
  otel-config.yaml: {{ toYaml ({{ include "otel-config-content" . ) | indent 4 }}
```